rust/lance-encoding/src/encodings/physical/binary.rs

```diff
@@ -739,13 +739,92 @@ impl MiniBlockDecompressor for BinaryMiniBlockDecompressor {
     }
 }
 
+#[derive(Debug, Default)]
+pub struct BinaryBlockEncoder {}
+impl BlockCompressor for BinaryBlockEncoder {
+    fn compress(&self, data: DataBlock) -> Result<LanceBuffer> {
+        let num_values: u32 = data
+            .num_values()
+            .try_into()
+            .expect("The Maximum number of values BinaryBlockEncoder can work with is u32::MAX");
+
+        match data {
+            DataBlock::VariableWidth(mut variable_width_data) => {
+                if variable_width_data.bits_per_offset != 32 {
+                    panic!("BinaryBlockEncoder only works with 32 bits per offset VariableWidth DataBlock.");
+                }
+                let offsets = variable_width_data.offsets.borrow_to_typed_slice::<u32>();
+                let offsets = offsets.as_ref();
+                // the first 4 bytes store the number of values, then 4 bytes for bytes_start_offset,
+                // then offsets data, then bytes data.
+                let bytes_start_offset = std::mem::size_of_val(offsets) as u32 + 4 + 4;
+
+                const BINARY_BLOCK_ALIGNMENT: usize = 4;
+                const BLOCK_PAD_BUFFER: [u8; BINARY_BLOCK_ALIGNMENT] = [72; BINARY_BLOCK_ALIGNMENT];
+
+                let output_total_bytes =
+                    ((bytes_start_offset as usize + variable_width_data.data.len()) + 3) / 4;
```

Comment:
I'm not sure I understand why the `+3 / 4`?

Reply:
yeah, this is a mistake, thanks!